src/AccessibilityInsights.SharedUx/Settings/ConfigurationModel.cs

```diff
@@ -652,5 +652,15 @@ public object Clone()
             return new ConfigurationModel(this);
         }
         #endregion
+
+        /// <summary>
+        /// Get version from AccessibilityInsights.SharedUx Assembly
+        /// </summary>
+        /// <returns></returns>
+        public static string GetAppVersionFromAssembly()
```

Comment:
>GetAppVersionFromAssembly [](start = 29, length = 25)

The ConfigurationModel class is intended to hold user config data. The assembly version is not user config data, and therefore doesn't belong in this class. There are many places where it might logically live. One easy option would be new static class in SharedUx.Misc.

Reply:
I think we still keep it in the settings folder, I don't like creating a separate utility classes as we used to do in Core to get the app version, cause thinking OOP, as you say there should be some class logically responsible for getting the assembly app version, so I think that the "ConfigurationManager" can do that and there's no need for a new class, what do you think?

Reply:
Agree with Dave that this shouldn't be in ConfigurationModel. Maybe ConfigurationManager makes more sense, but it still seems like this isn't "configuration"

Reply:
This isn't config. It's a static value that we're exposing. The "Single-responsibility principle" (the "S" in "SOLID") tells us that this belongs in a class with similar responsibilities. If we don't have one, we should create one.

---
In reply to: [276050404](https://github.com/Microsoft/accessibility-insights-windows/pull/309#discussion_r276050404) [](ancestors = 276050404)

Reply:
@DaveTryon  is right on the money here. This should be a separate class and definitely not part of the configuration manager. One way to tell this is not the right class for the job is to ask the question: why does the configuration manager class change. The answer: it changes when configuration options change. Classes change for one reason and one reason only. Therefore, version does not belong; because if the version logic changes, we should not change the configuration class.